crates/symbolicator-service/src/services/cacher.rs

```diff
@@ -694,6 +682,42 @@ mod tests {
         assert_eq!(request.computations.load(Ordering::SeqCst), 1);
     }
 
+    /// Makes sure that a `NotFound` result does not fall back to older cache versions.
+    #[tokio::test]
+    async fn test_cache_fallback_notfound() {
+        test::setup();
+
+        let cache_dir = test::tempdir().path().join("test");
+        std::fs::create_dir_all(cache_dir.join("global/1")).unwrap();
+        std::fs::write(
+            cache_dir.join("global/some_cache_key"),
+            "some old cached contents",
+        )
+        .unwrap();
+        std::fs::create_dir_all(cache_dir.join("1/global")).unwrap();
```

Comment:
Is there intentionally both `global/1` and `1/global`?

Reply:
good catch! I got confused first until I remembered that the cache key is `$version/$scope` and not the other way around :-)